### PR TITLE
feat(feedback): use backstage auth service in backend plugin

### LINKED ISSUE #1626: Feedback Plugin: 401 Unauthorized error
### Describe the bug

We are implementing the feedback on our backstage instance, but we are getting a 401 Unauthorized when the front end plugin is making API calls to the backend. I think this is happening because the API calls from the front end are not using backstage's fetchApi and is not including any auth headers.

### Expected Behavior
The frontend should not have issues making API calls to the backend even if backend auth is enabled.

### What are the steps to reproduce this bug?

1. Enable backend auth on backstage
2. Setup feedback plugin's backend according to the docs
3. Setup feedback plugin's frontend according to the docs
4. Run backstage and try to provide feedback through the plugin

### Versions of software used and environment
@janus-idp/backstage-plugin-feedback version 1.2.5
@janus-idp/backstage-plugin-feedback version 1.1.6
Backstage version 1.26.5

### PR DESCRIPTION
**Pull request by yashoswalyo**
- use backstage AuthService to get token for catalog api
- use backstage LoggerSevice
- use backstage FetchApi in frontend plugin ( fixes #1626 )
- updated tests

### CODE REVIEW

File: plugins/feedback-backend/README.md [hunk 1]
```diff
@@ -15,7 +15,20 @@ Install the NPM Package
 yarn workspace backend add @janus-idp/backstage-plugin-feedback-backend
 ```
 
-#### Adding the plugin to the legacy backend
+#### Adding the plugin to the new backend
+
+Add the following to your `packages/backend/src/index.ts` file:
+
+```ts title="packages/abckend/src/index.ts"
```
Comment (riginoommen) [inline plugins/feedback-backend/README.md:22]:
```suggestion
```ts title="packages/backend/src/index.ts"
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

**Review by riginoommen — APPROVED** [review]

**Review by deshmukhmayur — COMMENTED** [review]
LGTM👍

**Review by deshmukhmayur — APPROVED** [review]

**Comment by kadel** [comment]
/approve
approving it based on the review form @deshmukhmayur and @riginoommen

**Pull request merged**